clickhouse/datadog_checks/clickhouse/clickhouse.py

```diff
@@ -76,7 +76,12 @@ def validate_config(self):
         if not self._server:
             raise ConfigurationError('the `server` setting is required')
 
-    def create_connection(self):
+    def connect(self):
+        if self._client is not None:
+            # Already connected.
+            self.service_check(self.SERVICE_CHECK_CONNECT, self.OK, tags=self._tags)
+            return
```

Comment:
It seems that we don't recover or recreate the client in case the client become invalid for some reason. Can that happen ? If so, we might want to address that (not in PR). cc @ofek

Reply:
Yes, agreed that's a problem, I added a note about this "reconnect on issue" in the PR description.